src/com/zetcode/Ball.java

```diff
@@ -23,12 +29,33 @@ private void initBall() {
     }
 
     private void loadImage() {
+        InputStream stream = getClass().getResourceAsStream(Commons.BALL_IMAGE);
+        ImageIcon ii;
+        try {
+            ii = new ImageIcon(ImageIO.read(stream));
+            image = ii.getImage();
+        } catch (IOException ex) {
+            System.out.println("Unable to load ball.png");
+        }
+    }
 
-        var ii = new ImageIcon("src/resources/ball.png");
-        image = ii.getImage();
+    public static synchronized void playSound() {
```

Comment:
Instead of playing the sound directly in the ball, I would add a SoundPlayer to the game and inject that as a dependency of the game.  I would then call the sound player PlaySound from the location where when the ball impact is detected.  This would reduce the number of changes on one pull request.  There are too many coupled changes in this PR imo.